extension/main.js

```diff
@@ -29,24 +29,32 @@
 				}
 			}
 
+			if (interval !== 'undefined') {
+				period = 1;
```

Comment:
This is overwriting what was set above...  in every case except when `interval === 'undefined'` you overwrote what was determined for `period`.  What is the intention here?